### PR TITLE
fix(velocity): this ensures that whatever language the user is viewin…

### LINKED ISSUE #22534: Caching empty containers in the Velocity2 cache when accessing in a language where the content doesn't exist
**Describe the bug**
If you access a page in a language where the page and the content in the page both do not exist we are caching the empty container in the Velocity2 cache

Reproduced on: 21.06, 22.06

Related Ticket: https://dotcms.zendesk.com/agent/tickets/107703

**To Reproduce**
The environment must be configured with the defaults for the two following variables:

DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE=false
DEFAULT_PAGE_TO_DEFAULT_LANGUAGE=true

1. From the demo starter, create a page called 'test-page' in the root in English
2. Add a new Rich Text contentlet to the container on that page in English
NOTE: Do not add Spanish versions of either the page or the content
3. Clear the cache
4. In another browser/incognito, go to http://localhost:8080/test-page?language_id=2, you will correctly see an empty container
5. Now go to http://localhost:8080/test-page?language_id=1 and you will incorrectly see an empty container
6. Clear the Velocity2 cache and revisit http://localhost:8080/test-page?language_id=1, the Rich Text will properly display

It appears we are caching the alternate language versions of content into the fallback default page containers in the Velocity2 cache

**Expected behavior**
When falling back we should not cache the content into the default language container

**Additional context**
This only occurs when we fall back the page and not the content. If the page exists in both languages but the content does not, it is cached correctly. 
It als only occurs if the non-existent language is accessed before the default, otherwise the default language gets properly cached.

**WORKAROUNDS**
Create a version of the page in all languages
Set DEFAULT_CONTENT_TO_DEFAULT_LANGUAGE to true

### PR DESCRIPTION
**Pull request by wezell**
this ensures that whatever language the user is viewing the site in, that the pages will be cached separately.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java [hunk 1]
```diff
@@ -911,5 +910,45 @@ private IHTMLPage findPageInDefaultLanguageDifferentThanProvided(String identifi
...
 
+    @Override
+    public IHTMLPage findByIdLanguageVariantFallback(@NotNull String identifier, long tryLang,
+                                                     @NotNull String tryVariant, boolean live,
+                                                     @NotNull User user,
+                                                     boolean respectFrontEndPermissions) throws DotSecurityException {
+
+
+        long defaultLang = APILocator.getLanguageAPI().getDefaultLanguage().getId();
+        boolean fallbackLang = tryLang != defaultLang && Config.getBooleanProperty("DEFAULT_PAGE_TO_DEFAULT_LANGUAGE", true);
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/HTMLPageAssetAPIImpl.java:921]: the call to Config could be blocking, it is better to use a Lazy

**Review by jdotcms — APPROVED** [review]

**Review by jcastro-dotcms — APPROVED** [review]
